site/content/docs/terraform-provider/testing-scripts-locally.md

```diff
@@ -6,80 +6,203 @@ menu:
+Having the possibility to run Playwright Test scripts for your browser checks locally will allow you to develop them faster and with more confidence.
 
 ## Enabling local script execution
 
 You can easily enable local execution by storing your browser check scripts in local folders and passing the path to the right script:
 
-```terraform
+{{< tabs "TF Check Resource" >}}
+{{< tab "TypeScript" >}}
+ ```terraform
 resource "checkly_check" "e2e-checkout" {
   name                      = "Checkout Flow"
   type                      = "BROWSER"
   activated                 = true
   should_fail               = false
   frequency                 = 1
   double_check              = true
-  ssl_check                 = false
   use_global_alert_settings = true
   locations = [
     "us-west-1",
     "eu-central-1"
   ]
 
-  script = file("${path.module}/scripts/checkout.js") // Our script is contained in this file
+  script = file("${path.module}/__checks__/checkout.spec.ts") // Our script is contained in this file
 }
 ```
+{{< /tab >}}
+{{< tab "JavaScript" >}}
+ ```terraform
+resource "checkly_check" "e2e-checkout" {
+  name                      = "Checkout Flow"
+  type                      = "BROWSER"
+  activated                 = true
+  should_fail               = false
+  frequency                 = 1
+  double_check              = true
+  use_global_alert_settings = true
+  locations = [
+    "us-west-1",
+    "eu-central-1"
+  ]
 
-In order to have the script work both locally and on Checkly without changes, make sure your script is wrapped in an async function:
+  script = file("${path.module}/__checks__/checkout.spec.js") // Our script is contained in this file
+}
+ ```
+{{< /tab >}}
+{{< /tabs >}}
 
-```javascript
-async function run () {
-  const browser = await chromium.launch()
-  const page = await browser.newPage()
+Basic checks written with `@playwright/test` will run locally with `npx playwright test` and remotely on Checkly without any modifications: 
 
-  const response = await page.goto('https://danube-web.shop')
+{{< tabs "Basic Example" >}}
+{{< tab "TypeScript" >}}
+ ```ts
+import { test, expect } from '@playwright/test'
 
-  // ...
-  
-  await page.close()
-  await browser.close()
-}
+test('Should load the web store', async ({ page }) => {
+  await page.goto('https://danube-web.shop')
 
-run()
+  await expect(page).toHaveTitle('danube-store')
+})
 ```
+{{< /tab >}}
+{{< tab "JavaScript" >}}
+ ```js
+const { test, expect } = require('@playwright/test')
+
+test('Should load the web store', async ({ page }) => {
+  await page.goto('https://danube-web.shop')
+
+  await expect(page).toHaveTitle('danube-store')
+})
+ ```
+{{< /tab >}}
+{{< /tabs >}}
+
+If your script is using [Page Object Models](https://playwright.dev/docs/pom) or imports other files, you can take advantage of Checkly's code snippets. Consider the following directory structure:
+
+{{< tabs "Directory Structure" >}}
+{{< tab "TypeScript" >}}
+ ```
+__checks__ /
```

Comment:
@adamwardecki __checks__ as a directory is part of the CLI "approach" afaik, i.e. in [previous TF docs](https://checklyhq-iq95p63hq-checkly.vercel.app/docs/terraform-provider/checks-groups/) we always referred to a `scripts` (and `snippets`) folder. I think we should decide on a single approach and keep it consistent not to get users confused. I would opt to keep `scripts` or similar to differentiate from the CLI approach.

Reply:
@ragog I changed the dir structure to be: 

```
scripts /
|--- snippets /
|     |--- shoppingCart.ts
|--- checkout.spec.ts
```

I know we had `scripts` and `snippets` as siblings elsewhere. I did it this way because then we can do `import { ShoppingCart } from './snippets/shoppingCart'` without any changes in the paths both locally and in checkly. Let me know if this works - I'll then also make the changes in the "Snippets and environment variables" docs.

Reply:
@adamwardecki totally works. The benefits are clear.